src/main/java/org/candlepin/subscriptions/event/EventController.java

```diff
@@ -277,36 +285,58 @@ public boolean validateServiceInstanceEvent(Event event) {
       log.warn("Event measurement value(s) must be >= 0. event={}", event);
       return false;
     }
+    return true;
+  }
+
+  private boolean isValidProductTag(Event event) {
+    // If product tag is already present in the event then verify whether it is present in our
+    // config
+    if (Objects.nonNull(event.getProductTag()) && !event.getProductTag().isEmpty()) {
+      boolean isValid =
+          event.getProductTag().stream().findFirst().map(Variant::isValidProductTag).orElse(false);
+      if (!isValid) {
+        log.warn("Product tag {} is invalid.", event.getProductTag().stream().findFirst());
+      }
+      return isValid;
+    } else {
+      // Determine whether the product is payg or non-payg, and then add the appropriate tag in
+      // SWATCH-1993. We are only checking for payg at this time because we only support payg in
+      // this
+      // flow, and we don't have a way to distinguish between payg and non-payg through events.
+      String role = event.getRole() != null ? event.getRole().toString() : null;
```

Comment:
maybe if we can we do this check in filterOnApplicableTags(event, role)? So that we can just have event object.